src/xo-app/settings/servers/index.js

```diff
@@ -113,7 +113,7 @@ export default class Servers extends Component {
               <td>
                 <Text
                   value={server.host}
-                  onChange={host => editServer(server, { host })}
+                  onChange={host => (server.status !== 'connected') && editServer(server, { host })}
```

Comment:
I'm not opposed to having this change in xo-web but it should in priority be in xo-server.